### PR TITLE
[ITG-65] Task editing frontend

### PR DESCRIPTION
**Pull request by john-sonz**
Fronted for task editing

---

#### Resources

- [ITG-65](https://kulawska.atlassian.net/browse/ITG-65)

### CODE REVIEW

File: cabra/src/pages/TaskDetails.tsx [hunk 1]
```diff
@@ -1,36 +1,52 @@
...
+import NavigationButton from "./components/NavigationButton";
 import Task from "./components/Task";
 import axios from "axios";
+import { routeHelpers } from "../routes";
 import { useQuery } from "react-query";
 
 export default function TaskDetailsPage() {
   const navigate = useNavigate();
-  const { id } = useParams();
+  const { id } = useParams() as { id: string };
```
Comment (tgargula) [inline cabra/src/pages/TaskDetails.tsx:16]:
```suggestion
  const { id } = useParams<{ id: string }>();
```

File: cabra/src/pages/TaskEdit.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import { ArrowLeftIcon } from "@heroicons/react/solid";
+import { ITask } from "../types/task";
+import Layout from "./components/Layout";
+import NavigationButton from "./components/NavigationButton";
+import axios from "axios";
+import { routeHelpers } from "../routes";
+
+export default function TaskEditPage() {
+  const navigate = useNavigate();
+  const { id } = useParams();
```
Comment (tgargula) [inline cabra/src/pages/TaskEdit.tsx:16]:
```suggestion
  const { id } = useParams<{ id: string }>();
```

File: cabra/src/pages/TaskEdit.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+  const addTask = useMutation((task: TaskFormInputs) =>
+    axios.put<ITask>(`/tasks/${id}`, task)
+  );
+
+  const onSubmit = (variables: TaskFormInputs) =>
+    addTask.mutateAsync(variables, {
+      onSuccess: () => {
+        client.invalidateQueries("tasks");
+        client.invalidateQueries(["tasks", id]);
```
Comment (tgargula) [inline cabra/src/pages/TaskEdit.tsx:35]: Maybe we should extract React query keys to a constant? Wdyt?

File: cabra/src/pages/components/CreateTaskForm.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+export default function CreateTaskForm() {
+  const addTask = useMutation((task: TaskFormInputs) =>
+    axios.post<ITask>("/tasks", task)
+  );
+  const client = useQueryClient();
+
+  const onSubmit = (variables: TaskFormInputs) =>
+    addTask.mutateAsync(variables, {
+      onSuccess: () => {
+        client.invalidateQueries("tasks");
```
Comment (tgargula) [inline cabra/src/pages/components/CreateTaskForm.tsx:22]: Same here

**Review by tgargula — APPROVED** [review]
LFG!!! :rocket:

File: cabra/src/pages/TaskDetails.tsx (hunk 1 above)
Comment (john-sonz) [reply cabra/src/pages/TaskDetails.tsx:16]: That's not gonna work unfortunately 😢 The generic is getting `Partial`ized by `react-router` and `id` still comes out as `string | undefined` while I need it to be just `string`.

File: cabra/src/pages/TaskEdit.tsx (hunk 3 above)
Comment (john-sonz) [reply cabra/src/pages/TaskEdit.tsx:35]: Yeah, and we should probably gather them with related HTTP calls

**Pull request merged**